### PR TITLE
Graduate labs components to core

### PR DESCRIPTION
**Pull request by adidahiya**
Feature branch: DO NOT SQUASH

### CODE REVIEW

File: packages/core/src/common/errors.ts [hunk 1]
```diff
@@ -40,9 +39,11 @@ export const NUMERIC_INPUT_STEP_SIZE_NON_POSITIVE =
     ns + ` <NumericInput> requires stepSize to be strictly greater than zero.`;
 export const NUMERIC_INPUT_STEP_SIZE_NULL = ns + ` <NumericInput> requires stepSize to be defined.`;
 
+// TODO (clewis): Migrate old Popover validation errors to the component formerly known as Popover2.
```
Comment (cmslewis) [inline packages/core/src/common/errors.ts]: Stray TODO. It's done though; can delete.

File: packages/datetime/src/dateInput.tsx [hunk 2]
```diff
@@ -362,6 +361,10 @@ export class DateInput extends AbstractPureComponent<IDateInputProps, IDateInput
     };
 
     private handleInputClick = (e: React.SyntheticEvent<HTMLInputElement>) => {
+        // stop propagation to the Popover's internal handleTargetClick handler;
+        // otherwise, the popover will flicker closed as soon as it opens.
+        e.stopPropagation();
```
Comment (adidahiya) [inline packages/datetime/src/dateInput.tsx]: strange, how come this wasn't necessary before?

File: packages/core/test/menu/menuTests.tsx [hunk 3]
```diff
@@ -151,39 +151,42 @@ describe("MenuItem", () => {
...
                 assert.isNotNull(childContainer.query(`.${Classes.OVERLAY_OPEN}`));
                 assert.isNull(childContainer.query(`.${Classes.ALIGN_LEFT}`));
                 done();
             });
         });
 
-        it("children will continue displaying in the same direction if possible", done => {
-            menuItem = ReactDOM.render(
-                <MenuItem iconName="style" text="Style">
+        // TODO (clewis): Commented this out during the Popover2 => Popover refactor. Get this working again.
```
Comment (cmslewis) [inline packages/core/test/menu/menuTests.tsx]: Never got this test working, even though the functionality works. Should I try again now?

File: packages/core/test/menu/menuTests.tsx [hunk 4]
```diff
@@ -203,9 +205,11 @@ describe("MenuItem", () => {
             });
         });
 
-        it("children will flip direction after no more room in the existing direction", done => {
-            menuItem = ReactDOM.render(
-                <MenuItem iconName="style" text="Style">
+        // TODO (clewis): Commented this out during the Popover2 => Popover refactor. Get this working again.
```
Comment (cmslewis) [inline packages/core/test/menu/menuTests.tsx]: Same

File: yarn.lock [hunk 5]
```diff
@@ -114,7 +114,7 @@
   version "0.2.4"
   resolved "https://registry.yarnpkg.com/@types/tapable/-/tapable-0.2.4.tgz#8181a228da46185439300e600c5ae3b3b3982585"
 
-"@types/tether@^1.1.27", "@types/tether@^1.4.3":
+"@types/tether@^1.4.3":
```
Comment (adidahiya) [inline yarn.lock]: hmm, we should probably be able to drop this entry

File: packages/core/test/menu/menuTests.tsx (hunk 3 above)
Comment (adidahiya) [reply packages/core/test/menu/menuTests.tsx]: yes, but it shouldn't block this PR. please file a follow-up ticket and add it to the 2.0.0 milestone

File: packages/core/test/menu/menuTests.tsx (hunk 4 above)
Comment (adidahiya) [reply packages/core/test/menu/menuTests.tsx]: same

File: yarn.lock (hunk 5 above)
Comment (adidahiya) [reply yarn.lock]: done

File: packages/core/test/menu/menuTests.tsx (hunk 3 above)
Comment (cmslewis) [reply packages/core/test/menu/menuTests.tsx]: https://github.com/palantir/blueprint/issues/2008

File: packages/datetime/src/dateInput.tsx (hunk 2 above)
Comment (cmslewis) [reply packages/datetime/src/dateInput.tsx]: Not sure, but `DateRangeInput` already had something like this FWIW:
https://github.com/palantir/blueprint/blob/develop/packages/datetime/src/dateRangeInput.tsx#L620

**Comment by adidahiya** [comment]
Table test failure looks like a flake. I'll refactor the harness to use Enzyme harnesses in a separate PR.

I noticed a Popover regression -- the placement dropdown doesn't seem to do anything here

image

**Comment by cmslewis** [comment]
@adidahiya I'll take a look.

**Comment by adidahiya** [comment]
latest commits lgtm

**Comment by adidahiya** [comment]
looks great!! :shipit:

**Comment by cmslewis** [comment]
@adidahiya FWIW it's possible to get into a state in the example when `isOpen={true}` and the popover content is too big to let you toggle set the `isOpen` switch back to false:

image

**Comment by adidahiya** [comment]
you can probably work around that by getting a bigger monitor. or zooming out :P

**Pull request merged**